sematic/resolvers/cloud_resolver.py

```diff
@@ -72,18 +101,62 @@ def set_graph(self, runs: List[Run], artifacts: List[Artifact], edges: List[Edge
         self._artifacts = {artifact.id: artifact for artifact in artifacts}
         self._edges = {make_edge_key(edge): edge for edge in edges}
 
-    def _get_resolution_image(self) -> Optional[str]:
-        return get_image_uri()
+    @memoized_property
+    def _container_image_uris(self) -> Optional[Dict[str, str]]:
+        return get_image_uris()
+
+    def _get_container_image(self, future: AbstractFuture) -> Optional[str]:
+        if self._container_image_uris is None:
+            return None
+
+        if future.props.inline:
+            return self._get_resolution_container_image()
+
+        base_image_tag = future.props.base_image_tag or DEFAULT_BASE_IMAGE_TAG
+
+        return self._get_tagged_image(base_image_tag)
+
+    def _get_resolution_container_image(self) -> Optional[str]:
+        if not self._detach:
```

Comment:
Is this supposed to be `_detach` or `_detached`?
This could be a source of confusion.

Reply:
`_detach` is set by the `detach` instantiation argument. It signals the user's intent to detach resolution or not.
`_detached` is a rename of `_is_running_remotely` and is passed by `worker.py` to indicate that the resolution is actually detached.

Reply:
Yes, the meaning of the terms is clear. Here I was asking if the decision should be made based on the user intent or on the state of the Resolver.

Reply:
This is not changing any functionality, just renaming a variable.
`detach` is a user settings, `detached` is an undocumented internal setting that `worker.py` passes to the resolver.